library/analytics/src/main/java/org/cru/godtools/analytics/firebase/FirebaseAnalyticsService.kt

```diff
@@ -54,50 +56,41 @@ class FirebaseAnalyticsService @VisibleForTesting internal constructor(
     @MainThread
     @Subscribe(threadMode = ThreadMode.MAIN_ORDERED)
     fun onAnalyticsEvent(event: AnalyticsBaseEvent) {
-        if (event.isForSystem(AnalyticsSystem.FIREBASE)) when (event) {
+        if (event.isForSystem(AnalyticsSystem.FIREBASE) || event.isForSystem(AnalyticsSystem.ADOBE)) when (event) {
             is AnalyticsScreenEvent -> handleScreenEvent(event)
             is AnalyticsActionEvent -> handleActionEvent(event)
         }
     }
 
-    // region ActivityLifecycleCallbacks
-    init {
-        app.registerActivityLifecycleCallbacks(this)
-    }
-
-    private var currentActivity: Activity? by weak()
-
-    override fun onActivityResumed(activity: Activity) {
-        currentActivity = activity
-    }
-
-    override fun onActivityPaused(activity: Activity) {
-        currentActivity = currentActivity?.takeUnless { it === activity }
-    }
-
-    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) = Unit
-    override fun onActivityStarted(activity: Activity) = Unit
-    override fun onActivityStopped(activity: Activity) = Unit
-    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) = Unit
-    override fun onActivityDestroyed(activity: Activity) = Unit
-    // endregion ActivityLifecycleCallbacks
-    // endregion Tracking Events
-
     @MainThread
     private fun handleScreenEvent(event: AnalyticsScreenEvent) {
-        currentActivity?.let { firebase.setCurrentScreen(it, event.screen, null) }
+        val bundle = Bundle().apply {
+            putString(FirebaseAnalytics.Param.SCREEN_NAME, event.screen)
+            putString("cru_appname", VALUE_GODTOOLS)
+            event.locale?.let { putString("cru_contentlanguage", LocaleCompat.toLanguageTag(it)) }
+            event.adobeSiteSection?.let { putString("cru_sitesection", event.adobeSiteSection) }
+            event.adobeSiteSubSection?.let { putString("cru_sitesubsection", event.adobeSiteSubSection) }
+        }
+        firebase.logEvent(FirebaseAnalytics.Event.SCREEN_VIEW, bundle)
     }
 
     @MainThread
     private fun handleActionEvent(event: AnalyticsActionEvent) {
-        firebase.logEvent(event.firebaseEventName, null)
+        val bundle = Bundle().apply {
+            event.adobeAttributes?.forEach { attribute ->
+                val attributeKey = attribute.key.replace(Regex("[ \\-.]"), "_")
+                putString(attributeKey, attribute.value.toString())
+            }
+        }
+
+        firebase.logEvent(event.firebaseEventName, bundle)
     }
 
     init {
         oktaUserProfileProvider.userInfoFlow()
             .map { it?.ssoGuid }
             .distinctUntilChanged()
-            .onEach { firebase.setUserId(it) }
+            .onEach { firebase.setUserId(it); firebase.setUserProperty("debug", BuildConfig.DEBUG.toString()) }
```

Comment:
a couple things here.
1. try and avoid having multiple statements on a single line, you should have added line returns so that the 2 statements were individual lines.
2. you should just set the `debug` property directly in the init block, not every time the logged in user changes. See how `USER_PROP_APP_TYPE` is set a single time